config/receiver.go

```diff
@@ -16,10 +16,30 @@ package config
 
 // Receiver is the configuration of a receiver. Specific receivers must implement this
 // interface and will typically embed ReceiverSettings struct or a struct that extends it.
+// Embedded CustomConfigOptions will force each receiver to implement Validate() function
 type Receiver interface {
 	NamedEntity
+	CustomConfigOptions
 }
 
+// CustomConfigOptions defines the interfaces for configuration customization options
+// Different custom interfaces can be added like CustomConfigValidator, CustomConfigUnmarshaller, etc.
+type CustomConfigOptions interface {
+	CustomConfigValidator
+	// CustomConfigUnmarshaller
+}
```

Comment:
From the discussion above we only want validation to be required since custom unmarshaling is more of an edge case so I would remove the references for now

```suggestion
// Different custom interfaces can be added like CustomConfigValidator, etc.
type CustomConfigOptions interface {
	CustomConfigValidator
}
```

Reply:
agree. updated